pkg/utils/helpers.go

```diff
@@ -115,6 +161,100 @@ func AddSecretFieldsToObj(razeeData map[string][]byte) (map[string]string, error
+
+	newField := []byte{}
+	var razeeStruct *marketplacev1alpha1.RazeeConfigurationValues = &marketplacev1alpha1.RazeeConfigurationValues{}
+	var error error
+	keys := []string{}
+	for key, element := range razeeData {
+		keys = append(keys, key)
+		value, err := RetrieveSecretField(element)
+		if err != nil {
+			error = err
+		}
+
+		// fullfill the SecretKeySelector struct
+		if key == "IBM_COS_READER_KEY" {
+			newField = []byte(fmt.Sprintf(
+				`{"%v": {"name": "rhm-operator-secret","key": "%v"}}`, key, key))
+		} else if key == "RAZEE_DASH_ORG_KEY" {
+			newField = []byte(fmt.Sprintf(
+				`{"%v": {"name": "rhm-operator-secret","key": "%v"}}`, key, key))
+		} else {
+			newField = []byte(fmt.Sprintf(`{"%v": "%v"}`, key, value))
+		}
+
+		// add to the struct
+		err = json.Unmarshal(newField, &razeeStruct)
+		if err != nil {
+			error = err
+		}
+		if err != nil {
+			error = err
+		}
+	}
+
+	missingItems := ContainsMultiple(keys, GetStructFieldsOnBase(marketplacev1alpha1.RazeeConfigurationValues{}))
+	return razeeStruct, missingItems, error
+}
+
+// returns the fields on the type/base struct so we don't to maintain a list of all the fields
+func GetStructFieldsOnBase(razeeStruct marketplacev1alpha1.RazeeConfigurationValues) []string {
+	aa := reflect.Indirect(reflect.ValueOf(razeeStruct))
+	fieldSlice := []string{}
+	for i := 0; i < aa.NumField(); i++ {
+		field := aa.Type().Field(i)
+		if jsonTag := field.Tag.Get("json"); jsonTag != "" && jsonTag != "-" {
+			if commaIdx := strings.Index(jsonTag, ","); commaIdx > 0 {
+				fieldName := jsonTag[:commaIdx]
+				fieldSlice = append(fieldSlice, string(fieldName))
+			}
+		}
+	}
+	return fieldSlice
+}
+
+func ConvertSecretToStruct(razeeData map[string][]byte) (marketplacev1alpha1.RazeeConfigurationValues, []string, error) {
+
+	newField := []byte{}
+	var razeeStruct *marketplacev1alpha1.RazeeConfigurationValues = &marketplacev1alpha1.RazeeConfigurationValues{}
+	var error error
+	keys := []string{}
+	for key, element := range razeeData {
+		keys = append(keys, key)
+		value, err := RetrieveSecretField(element)
+		if err != nil {
+			error = err
+		}
+
+		// fullfill the SecretKeySelector struct
+		if key == "IBM_COS_READER_KEY" {
+			newField = []byte(fmt.Sprintf(
+				`{"%v": {"name": "rhm-operator-secret","key": "%v"}}`, key, key))
+		} else if key == "RAZEE_DASH_ORG_KEY" {
+			newField = []byte(fmt.Sprintf(
+				`{"%v": {"name": "rhm-operator-secret","key": "%v"}}`, key, key))
+		} else {
+			newField = []byte(fmt.Sprintf(`{"%v": "%v"}`, key, value))
+		}
+
+		// add to the struct
+		err = json.Unmarshal(newField, &razeeStruct)
+		if err != nil {
+			fmt.Println("AddSecretFieldsToStruct", err)
+			error = err
+		}
+		if err != nil {
+			error = err
+		}
+	}
+
+	missingItems := ContainsMultiple(keys, GetStructFieldsOnBase(marketplacev1alpha1.RazeeConfigurationValues{}))
+	return *razeeStruct, missingItems, error
```

Comment:
Let's simplify this and just set the struct directly. The struct is hard coded and can not be returned any other way so dynamic code isn't really being used:

```go
obj := &RazeeConfigValues{}
switch key {
case IBM_COS_READER_KEY: obj.IbmCosReaderkey = value // whatever obj you need to set it to
//etc...
}
```

You can use a top level switch to set different fields.